server/etcdserver/corrupt.go

```diff
@@ -258,57 +259,152 @@ func (cm *corruptionChecker) CompactHashCheck() {
 	)
 	hashes := cm.uncheckedRevisions()
 	// Assume that revisions are ordered from largest to smallest
-	for i, hash := range hashes {
+	for _, hash := range hashes {
 		peers := cm.hasher.PeerHashByRev(hash.Revision)
```

Comment:
Improvements for the backlog:
1. getPeerHashKVs() could fetch the hashes in paralle.
2. `ServeHTTP` could populate list of hashes we have (for consumption in v3.7, 2028)

Reply:
For v3.6 release we should consider hash being negotiated via raft.

Reply:
Let's discuss & address this in a separate session/PR.  Renaming and fetching the hashes in parallel make sense to me. @ptabor  where & how is the `2028` coming from? :) 

For @serathius 's comment "`consider hash being negotiated via raft`", I did not get the point. My immediate feeling there is no need, because the compaction is already coordinated by raft.

Reply:
> we should consider hash being negotiated via raft.

Is it going to persist the hash result?

Reply:
> how is the 2028 coming from ?  

Just extrapolation or release frequency

> Rafting hashes ?

I imagine this could work that way that way: 
  1)  there is new RAFT message 'start-checksum' that triggers all members to compute checksum at the exact revision. So it's 'simpler' compaction.    Compaction stays as doing this implicitly. 
  2) Whenever member finishes the computation it sends to leader their result (pair: rev, hash). 
  3) Leader broadcasts ? the received results through RAFT
  4) Every-member can react on discrepancy. 

Benefit: Does not require custom service and best-effort attempts to check whether we have checks-in-sync. 

But I would consider evaluating (on-line) merkle root ( https://github.com/etcd-io/etcd/issues/13839 ) sums design first and thinking what raft changes would be needed for this.

Reply:
Thanks @ptabor for the comment.  It's a big topic, let's discuss it separately.